sdk/core/core/CMakeLists.txt

```diff
@@ -4,13 +4,32 @@
 cmake_minimum_required (VERSION 3.10)
 set(TARGET_NAME "az_core")
 
-project (${TARGET_NAME} LANGUAGES C)
+if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/version.txt)
+    file (STRINGS ${CMAKE_CURRENT_SOURCE_DIR}/version.txt VERSION_STRING)
+    message(STATUS "Read version from version.txt: " ${VERSION_STRING})
+    string(REGEX MATCHALL "([0-9]+).([0-9]+).([0.9]+)(.*)" PRE_RELEASE ${VERSION_STRING})
 
+    set(AZ_VERSION_MAJOR ${CMAKE_MATCH_1} )
+    set(AZ_VERSION_MINOR ${CMAKE_MATCH_2} )
+    set(AZ_VERSION_PATCH ${CMAKE_MATCH_3} )
+    if(CMAKE_MATCH_4)
+        string(SUBSTRING ${CMAKE_MATCH_4} 1 -1 PRERELEASE_SUBSTRING )
+        set(AZ_VERSION_PRERELEASE ${PRERELEASE_SUBSTRING})
+    endif()
+    message(STATUS "AZ_VERSION_MAJOR " ${AZ_VERSION_MAJOR})
+    message(STATUS "AZ_VERSION_MINOR " ${AZ_VERSION_MINOR})
+    message(STATUS "AZ_VERSION_PATCH " ${AZ_VERSION_PATCH})
+    message(STATUS "AZ_VERSION_PRERELEASE " ${AZ_VERSION_PRERELEASE})
```

Comment:
Should we only print this if `CMAKE_MATCH_4` exists, so move this in the if block above?

Reply:
I like seeing it always.  These are for debug purposes.  Its nice to see that it isn't set to anything.